ext/dom/element.c

```diff
@@ -960,14 +1098,21 @@ PHP_METHOD(DOMElement, removeAttributeNS)
 
 	DOM_GET_OBJ(nodep, id, xmlNodePtr, intern);
 
+	bool follow_spec = php_dom_follow_spec_intern(intern);
+	if (follow_spec && uri_len == 0) {
+		uri = NULL;
+	}
+
 	attrp = xmlHasNsProp(nodep, (xmlChar *)name, (xmlChar *)uri);
```

Comment:
Should those use `BAD_CAST` or not?

Reply:
Yeah, well this is untouched code in old DOM basically so that's why I didn't refactor this.

Reply:
ACK